src/legacy-routes.js

```diff
@@ -80,25 +152,30 @@ const addressSummary = (dbApi: any, { logger }: ServerConfig) => async (req: any
 const txSummary = (dbApi: any, { logger }: ServerConfig) => async (req: any,
 ) => {
   const { tx } = req.params
-  const result = await dbApi.txSummary(tx)
-  if (result.rows.length === 0) {
-    return { Left: invalidTx }
-  }
-  const row = result.rows[0]
-  const totalInput = arraySum(row.inputs_amount)
-  const totalOutput = arraySum(row.outputs_amount)
+  const { rows: txResult } = await dbApi.getTx(wrapHashPrefix(tx))
+  if (txResult.length === 0) return { Left: invalidTx }
+
+  const txRow = txResult[0]
+  const { rows: blockResult } = await dbApi.getBlockById(txRow.block)
+  const blockRow = blockResult[0]
+
+  const { rows: inputs } = await dbApi.getTxInputs(txRow.id)
+  const { rows: outputs } = await dbApi.getDistinctTxOutputs([txRow.id])
+
+  const totalInput = arraySum(inputs.map(elem => elem.value))
+  const totalOutput = arraySum(outputs.map(elem => elem.value))
   const epoch0 = 1506203091
   const slotSeconds = 20
   const epochSlots = 21600
-  const blockTime = moment(row.time).unix()
+  const blockTime = moment(blockRow.time).unix()
   const right = {
-    ctsId: row.hash,
+    ctsId: unwrapHashPrefix(txRow.hash),
     ctsTxTimeIssued: blockTime,
     ctsBlockTimeIssued: blockTime,
-    ctsBlockHeight: Number(row.block_num),
+    ctsBlockHeight: Number(blockRow.block_no),
     ctsBlockEpoch: Math.floor((blockTime - epoch0) / (epochSlots * slotSeconds)),
```

Comment:
I know this is old code, but isn't there a way to get the epoch/slot explicitly from the DB. I believe this was coded under the (erroneus) assumption that a new slot must appear each 20 seconds and was calibrated to match that at the time it was coded, so I wouldn't be surprised if this formula is not correct anymore